lib/charms/opensearch/v0/opensearch_relation_peer_cluster.py

```diff
@@ -376,18 +395,25 @@ def _s3_credentials(
                 return None
 
             # As the main orchestrator, this application must set the S3 information.
+            access_key = self.charm.backup.s3_client.get_s3_connection_info().get("access-key")
+            secret_key = self.charm.backup.s3_client.get_s3_connection_info().get("secret-key")
+
+            # set the secrets in the charm
+            self.charm.secrets.put(Scope.APP, "s3-access-key", access_key)
+            self.charm.secrets.put(Scope.APP, "s3-secret-key", secret_key)
```

Comment:
We should revisit and reformat this code once s3 relation starts supporting relation secrets. Can you add a `# TODO` so we can come back and remove this secret?